### PR TITLE
testnet scripts refactoring merging

### PR DESCRIPTION
**Pull request by dzmitryhil**

### CODE REVIEW

File: deploy/testnet/set-ulimit-hrel8.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
+# Set correct ulimits for hrel8
 
-Run the following line to set correct ulimits. 
+Run the following line to set correct ulimits.
```
Comment (Parth4337) [inline deploy/testnet/set-ulimit-hrel8.md]: Run the following command...

File: deploy/testnet/ethereum.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+* Install dependencies from source code
+
+```
+./bin.sh
+```
+
+* Start the node
+
+Before run the script please set up "ulimit > 65535" ([example-hrel8](set-ulimit-hrel8.md))
```
Comment (Parth4337) [inline deploy/testnet/ethereum.md]: hrel -> rhel (Red Hat Enterprise Linux)

File: deploy/testnet/ethereum.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+* Setup auto-start
+
+Add to your crontab or /etc/init.d scripts:
+
+* `start-geth.sh`
+* `start-node-exporter.sh`
+
+***If you used the bin.sh installation and what to use the scripts for the auto-start, additionally you need to
```
Comment (Parth4337) [inline deploy/testnet/ethereum.md]: If you used the bin.sh installation and **want** to use the scripts for the auto-start........

File: deploy/testnet/full.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+./bin.sh
+```
+
+* Init full node
+
+```
+./init-full-node.sh
+```
+
+* Expose monitoring
```
Comment (Parth4337) [inline deploy/testnet/full.md]: Isn't this optional? if yes, then it should be mentioned as well.

File: deploy/testnet/full.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+./expose-metrics.sh
+```
+
+* Optionally allow cors requests
+
+```
+./allow-cors.sh
+```
+
+* Run node exporter
```
Comment (Parth4337) [inline deploy/testnet/full.md]: Same as expose_metrics, if it is optional, it should be mentioned

File: deploy/testnet/full.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+```
+./start-node-exporter.sh
+```
+
+* Start the node
+
+Before run the script please set up "ulimit > 65535" ([example-hrel8](set-ulimit-hrel8.md))
```
Comment (Parth4337) [inline deploy/testnet/full.md]: hrel -> rhel

File: deploy/testnet/full.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+* Setup auto-start
+
+Add to your crontab or /etc/init.d scripts:
+
+* `start-node.sh`
+* `start-node-exporter.sh`
+
+***If you used the bin.sh installation and what to use the scripts for the auto-start, additionally you need to
```
Comment (Parth4337) [inline deploy/testnet/full.md]: *If you used the bin.sh installation and **want** to use the scripts....

File: deploy/testnet/seed.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+```
+./bin.sh
+```
+
+* Init seed node
+
+```
+./init-seed-node.sh
+```
+If other seeds were provided then the genesis.json was downloaded, if not then get the genesys file
```
Comment (Parth4337) [inline deploy/testnet/seed.md]: if not then download genesis file manually and replace it with $ONOMY_HOME/config/genesis.json

File: deploy/testnet/sentry.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+./start-node-exporter.sh
+```
+
+* Start the node
+
+Before run the script please set up "ulimit > 65535" ([example-hrel8](set-ulimit-hrel8.md))
+
+```
+./start-node.sh
+```
```
Comment (Parth4337) [inline deploy/testnet/sentry.md]: Duplicate

File: deploy/testnet/sentry.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+* Expose monitoring
+
+```
+./expose-metrics.sh
+```
+
```
Comment (Parth4337) [inline deploy/testnet/sentry.md]: Duplicate

File: deploy/testnet/set-ulimit-hrel8.md [hunk 11]
```diff
@@ -1,5 +1,6 @@
+# Set correct ulimits for hrel8
```
Comment (Parth4337) [inline deploy/testnet/set-ulimit-hrel8.md]: hrel -> rhel

File: deploy/testnet/validator.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+
+* Init full node
+
+```
+./init-full-node.sh
+```
+
+* Optionally with sentries
+
+    * Start sentry nodes based on instructions from the [sentry](../sentry/readme.md)
```
Comment (Parth4337) [inline deploy/testnet/validator.md]: from the [**sentry node setup**](../sentry/readme.md)

File: deploy/testnet/validator.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+        
+        Get the node ip:
+        
+        ```
+        hostname -I | awk '{print $1}'
+        ```
+        
+    * Run script to set up the private connection of the validator and sentries
+
+    You will need to provide the sentry IPs.
```
Comment (Parth4337) [inline deploy/testnet/validator.md]: "You will need to provide the sentry IPs." --> "Make sure to setup and start all the sentries before running this script as you will need to provide IPs of all the sentry nodes"

Also fix indentation

File: deploy/testnet/validator.md [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+* Expose onomy node monitoring
+
+```
+./expose-metrics.sh
+```
+
+* Start the node
+
+Before run the script please set up "ulimit > 65535" ([example-hrel8](set-ulimit-hrel8.md))
```
Comment (Parth4337) [inline deploy/testnet/validator.md]: Before **running** the script

File: deploy/testnet/validator.md [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+onomyd q bank balances {orchestrator-address}
+```
+
+* Init gbt
+
+```
+./init-gbt.sh
+```
+
+Check that your Ethereum address is in the list of curren valset
```
Comment (Parth4337) [inline deploy/testnet/validator.md]: typo: curren valset -> current valset

File: deploy/testnet/validator.md [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+```
+onomyd q gravity current-valset
+```
+
+* Run orchestrator
+
+Before run the script please set env variable:
+
+* ETH_RPC_ADDRESS - the RPC address of the Ethereum node
+
```
Comment (Parth4337) [inline deploy/testnet/validator.md]: Indent in, it should be a sub point of *Run Orchestrator*

Before running the script please set environment variable:

File: deploy/testnet/scripts/bin.sh [hunk 17]
```diff
@@ -22,6 +22,8 @@ sudo subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpm
 sudo dnf -y --skip-broken install curl nano ca-certificates tar git jq gcc-c++ gcc-toolset-9 openssl-devel musl-devel musl-gcc gmp-devel perl python3 moreutils wget nodejs make hostname procps-ng
 source "/opt/rh/gcc-toolset-9/enable"
 
+set -eu
+
 echo "--------------installing rust---------------------------"
 curl https://sh.rustup.rs -sSf | bash -s -- -y
 export PATH=$HOME/.cargo/bin:$PATH
```
Comment (Parth4337) [inline deploy/testnet/scripts/bin.sh:29]: Line 32-36 "Installing GO Lang"
We should cleanup after install -> `rm go.tar.gz`

File: deploy/testnet/scripts/bin.sh [hunk 18]
```diff
@@ -48,19 +50,17 @@ rustup target add x86_64-unknown-linux-musl
...
 git clone -b $GETH_VERSION https://github.com/ethereum/go-ethereum $GETH_SRC
 cd $GETH_SRC
 make geth
 cp build/bin/geth $ONOMY_HOME/bin/geth
 
+echo "-------------------installing node_exporter-----------------------"
+curl -LO "https://github.com/prometheus/node_exporter/releases/download/v$NODE_EXPORTER_VERSION/node_exporter-$NODE_EXPORTER_VERSION.linux-amd64.tar.gz"
+tar -xvf "node_exporter-$NODE_EXPORTER_VERSION.linux-amd64.tar.gz"
+mv "node_exporter-$NODE_EXPORTER_VERSION.linux-amd64/node_exporter" $ONOMY_HOME/bin/node_exporter
+
```
Comment (Parth4337) [inline deploy/testnet/scripts/bin.sh]: Clean up after install -> `rm node_exporter-$NODE_EXPORTER_VERSION.linux-amd64.tar.gz` and `rm -r "node_exporter-$NODE_EXPORTER_VERSION.linux-amd64`

File: deploy/testnet/scripts/init-master.sh [hunk 19]
```diff
@@ -6,8 +6,6 @@ echo "Initializing master node"
 ONOMY_HOME=$HOME/.onomy
```
Comment (Parth4337) [inline deploy/testnet/scripts/init-master.sh]: Why init-master is in here? We have moved it to environments haven't we?

File: deploy/testnet/scripts/init-validator.sh [hunk 20]
```diff
@@ -38,15 +36,9 @@ then
...
-   # The name of onomy orchestrator
-  read -r -p "Enter a name for your orchestrator [orchestrator]:" ONOMY_ORCHESTRATOR_NAME
-  ONOMY_ORCHESTRATOR_NAME=${ONOMY_ORCHESTRATOR_NAME:-orchestrator}
-  $ONOMY keys add $ONOMY_ORCHESTRATOR_NAME $ONOMY_KEYRING_FLAG --output json | jq . >> $ONOMY_HOME/orchestrator_key.json
-done
-
-
+read -r -p "Enter a name for your orchestrator [orchestrator]:" ONOMY_ORCHESTRATOR_NAME
+ONOMY_ORCHESTRATOR_NAME=${ONOMY_ORCHESTRATOR_NAME:-orchestrator}
+onomyd keys add $ONOMY_ORCHESTRATOR_NAME $ONOMY_KEYRING_FLAG --output json | jq . >> $ONOMY_HOME/orchestrator_key.json
```
Comment (Parth4337) [inline deploy/testnet/scripts/init-validator.sh:41]: Line 55 & Line 59
We should format output and highlight the address part to make it stand out
```
TEXTFG=$(tput setaf 2)
BOLD=$(tput bold)
UNDERLINE=$(tput smul)
NORM=$(tput sgr0)


echo -e "Your validator address is ${TEXTFG}${BOLD}${UNDERLINE}$ONOMY_VALIDATOR_ADDRESS${NORM}"
echo -e "Your orchestrator address is ${TEXTFG}${BOLD}${UNDERLINE}$ONOMY_ORCHESTRATOR_ADDRESS${NORM}"
```

File: deploy/testnet/set-ulimit-hrel8.md (hunk 1 above)
Comment (dzmitryhil) [reply deploy/testnet/set-ulimit-hrel8.md:3]: updated

File: deploy/testnet/ethereum.md (hunk 2 above)
Comment (dzmitryhil) [reply deploy/testnet/ethereum.md]: Updated

File: deploy/testnet/full.md (hunk 4 above)
Comment (dzmitryhil) [reply deploy/testnet/full.md]: updated to "Optionally expose monitoring"

File: deploy/testnet/full.md (hunk 5 above)
Comment (dzmitryhil) [reply deploy/testnet/full.md]: Updated -> "Optionally run node exporter"

File: deploy/testnet/full.md (hunk 6 above)
Comment (dzmitryhil) [reply deploy/testnet/full.md]: Update to full name -> "Red Hat Enterprise Linux"

File: deploy/testnet/full.md (hunk 7 above)
Comment (dzmitryhil) [reply deploy/testnet/full.md]: findex

File: deploy/testnet/scripts/bin.sh (hunk 17 above)
Comment (dzmitryhil) [reply deploy/testnet/scripts/bin.sh:29]: updated

File: deploy/testnet/scripts/bin.sh (hunk 18 above)
Comment (dzmitryhil) [reply deploy/testnet/scripts/bin.sh]: updated

File: deploy/testnet/scripts/init-master.sh (hunk 19 above)
Comment (dzmitryhil) [reply deploy/testnet/scripts/init-master.sh]: removed

File: deploy/testnet/scripts/init-validator.sh (hunk 20 above)
Comment (dzmitryhil) [reply deploy/testnet/scripts/init-validator.sh:41]: I tried with this output, as for me, looks the same. So let's keep the simple output.

File: deploy/testnet/seed.md (hunk 8 above)
Comment (dzmitryhil) [reply deploy/testnet/seed.md]: updated

File: deploy/testnet/sentry.md (hunk 10 above)
Comment (dzmitryhil) [reply deploy/testnet/sentry.md]: removed

File: deploy/testnet/sentry.md (hunk 9 above)
Comment (dzmitryhil) [reply deploy/testnet/sentry.md]: removed

File: deploy/testnet/set-ulimit-hrel8.md (hunk 11 above)
Comment (dzmitryhil) [reply deploy/testnet/set-ulimit-hrel8.md]: updated

File: deploy/testnet/set-ulimit-hrel8.md (hunk 1 above)
Comment (dzmitryhil) [reply deploy/testnet/set-ulimit-hrel8.md]: updated

File: deploy/testnet/validator.md (hunk 12 above)
Comment (dzmitryhil) [reply deploy/testnet/validator.md]: updated

File: deploy/testnet/validator.md (hunk 13 above)
Comment (dzmitryhil) [reply deploy/testnet/validator.md]: updated

File: deploy/testnet/validator.md (hunk 14 above)
Comment (dzmitryhil) [reply deploy/testnet/validator.md]: updated

File: deploy/testnet/validator.md (hunk 15 above)
Comment (dzmitryhil) [reply deploy/testnet/validator.md]: updated

File: deploy/testnet/validator.md (hunk 16 above)
Comment (dzmitryhil) [reply deploy/testnet/validator.md]: updated

File: deploy/testnet/ethereum.md (hunk 3 above)
Comment (dzmitryhil) [reply deploy/testnet/ethereum.md]: updated

File: deploy/monitoring/readme.md [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+```
+./start-monitoring.sh
+```
+
+The script will ask you to provide the node ip, use the IP of your node.
+
+If the monitoring is set up correctly you will see the output:
+
+```
+echo "Open the http://localhost:3000/ to access the Grafana, default credentials are: admin:admin"
```
Comment (Parth4337) [inline deploy/monitoring/readme.md:16]: remove echo just the output "Open the ......admin:admin"

**Review by Parth4337 — APPROVED** [review]

**Pull request merged**